lib/queue_classic/conn_adapter.rb

```diff
@@ -55,6 +55,11 @@ def server_version
                           end
     end
 
+    def active?
+      !@connection.finished? &&
+        @connection.transaction_status < PG::Connection::PQTRANS_INERROR
```

Comment:
Probably should be negation here
```suggestion
        !(@connection.transaction_status < PG::Connection::PQTRANS_INERROR)
```

Reply:
There's also exists `PQTRANS_UNKNOWN`, not sure if this value can happen normally.

Reply:
Here I was going for "is the connection active and still healthy" - which is everything this side of the INERROR enum.

So, `!finished and status < error` should be fine?